http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/LeoAppServiceInterp.scala

```diff
@@ -815,14 +834,47 @@ final class LeoAppServiceInterp[F[_]: Parallel](config: AppServiceConfig,
           app.appName,
           workspaceId,
           cloudContext,
-          diskOpt,
+          diskIdOpt,
           BillingProfileId(workspaceDesc.spendProfile),
           Some(ctx.traceId)
         )
         _ <- publisherQueue.offer(deleteMessage)
       } yield ()
   } yield ()
 
+  private def checkIfSubresourcesDeletable(appId: AppId,
+                                           resourceType: WsmResourceType,
+                                           userInfo: UserInfo,
+                                           workspaceId: WorkspaceId
+  )(implicit
+    ev: Ask[F, AppContext]
+  ): F[Unit] = for {
+    ctx <- ev.ask
+    wsmResources <- appControlledResourceQuery
+      .getAllForAppByType(appId.id, resourceType)
+      .transaction
+    _ <- wsmResources.traverse { resource =>
+      for {
+        wsmState <- resourceType match {
+          case WsmResourceType.AzureDatabase =>
+            wsmClientProvider.getDatabaseState(userInfo.accessToken.token, workspaceId, resource.resourceId)
+          case WsmResourceType.AzureKubernetesNamespace =>
+            wsmClientProvider.getNamespaceState(userInfo.accessToken.token, workspaceId, resource.resourceId)
+          case WsmResourceType.AzureManagedIdentity =>
+            wsmClientProvider.getIdentityState(userInfo.accessToken.token, workspaceId, resource.resourceId)
+          case WsmResourceType.AzureDisk =>
+            wsmClientProvider.getDiskState(userInfo.accessToken.token, workspaceId, resource.resourceId)
+          case WsmResourceType.AzureStorageContainer =>
+            F.pure(WsmState(None)) // no get endpoint for a storage container in WSM yet
+        }
+        _ <- F
+          .raiseUnless(wsmState.isDeletable)(
+            AppResourceCannotBeDeletedException(resource.resourceId, appId, wsmState.value, resourceType, ctx.traceId)
+          )
+      } yield ()
+    }
+  } yield ()
+
```

Comment:
see comments above re: my trying to understand more deeply why we need to do this - thank you for helping me understand more!